src/lib.rs

```diff
@@ -304,4 +326,56 @@ mod test {
 
         assert_eq!(1, Arc::try_unwrap(drop_count).unwrap().into_inner());
     }
+
+    #[test]
+    fn unwrap_ok() {
+        let drop_count = Arc::new(AtomicUsize::new(0));
+
+        let ok = vec![
+            DropCounter(drop_count.clone()),
+            DropCounter(drop_count.clone()),
+            DropCounter(drop_count.clone()),
+        ]
+        .into_iter()
+        .collect::<CollectArrayResult<_, 3>>()
+        .unwrap();
+        drop(ok);
+
+        assert_eq!(3, Arc::try_unwrap(drop_count).unwrap().into_inner());
+    }
+
+    #[test]
+    fn unwrap_too_many() {
+        let drop_count = Arc::new(AtomicUsize::new(0));
+
+        let ok = panic::catch_unwind(|| {
+            vec![
+                DropCounter(drop_count.clone()),
+                DropCounter(drop_count.clone()),
+                DropCounter(drop_count.clone()),
+            ]
+            .into_iter()
+            .collect::<CollectArrayResult<_, 1>>()
+            .unwrap()
+        });
+
+        assert!(ok.is_err());
+
+        assert_eq!(3, Arc::try_unwrap(drop_count).unwrap().into_inner());
+    }
+
+    #[test]
+    fn unwrap_not_enough() {
+        let drop_count = Arc::new(AtomicUsize::new(0));
+
+        let ok = panic::catch_unwind(|| {
+            vec![DropCounter(drop_count.clone())]
+                .into_iter()
+                .collect::<CollectArrayResult<_, 3>>()
+                .unwrap()
+        });
+        drop(ok);
+
+        assert_eq!(1, Arc::try_unwrap(drop_count).unwrap().into_inner());
```

Comment:
```suggestion
        
        assert!(ok.is_err());

        assert_eq!(1, Arc::try_unwrap(drop_count).unwrap().into_inner());
```

I suspect this was a mis-copy&paste - the drop of the elements here should be happening automatically, but we want to make sure we returned an error.